src/client/ua_client_highlevel_subscriptions.c

```diff
@@ -510,4 +568,51 @@ UA_Client_Subscriptions_clean(UA_Client *client){
         UA_Client_Subscriptions_forceDelete(client, sub); /* force local removal */
 }
 
+UA_StatusCode
+UA_Client_Subscriptions_backgroundPublish(UA_Client *client) {
+    if (client->state < UA_CLIENTSTATE_SESSION)
+        return UA_STATUSCODE_BADSERVERNOTCONNECTED;
+
+    if (client->config.outStandingPublishRequests == 0)
```

Comment:
There should also be a check if the client has any subscriptions. If there are no subscriptions, the server gets spammed with publish requests when UA_Client_runAsync() is called.

Reply:
Thanks. Done.